ciao-controller/command.go

```diff
@@ -266,65 +306,29 @@ func (c *controller) startWorkload(w types.WorkloadRequest) ([]*types.Instance,
 		if w.Subnet == "" {
 			newIP = IPPool[i]
 		}
+
+		name := w.Name
+		if name != "" {
+			if w.Instances > 1 {
+				name = fmt.Sprintf("%s-%d", name, i)
+			}
+		}
+
 		wg.Add(1)
-		go func(newIP net.IP) {
+		go func(newIP net.IP, name string) {
 			sem <- 1
 			defer func() {
 				<-sem
 				wg.Done()
 			}()
 
-			startTime := time.Now()
-
-			name := w.Name
-			if name != "" {
-				if w.Instances > 1 {
-					name = fmt.Sprintf("%s-%d", name, i)
-				}
-			}
-
-			instance, err := newInstance(c, w.TenantID, &wl, w.Volumes, name, w.Subnet, newIP)
+			instance, err := c.createInstance(w, wl, name, newIP)
```

Comment:
w and wl are effectively constant here aren't they in which case this is fine and there's no race.

Reply:
yes, they don't get modified.